File: CS/PS/Site/programmers/lv1/소수만들기.js
//서로다른 3개 뽑기 => how? 조합
function isPrime(num){
  const sqrt = Math.sqrt(num);
  for (let i = 2; i <= sqrt; i++){
    if(num % i == 0){
      return false;
    }
  }
  return true;
}

function solution(nums) {
  var answer = 0;
  const len = nums.length;
  for(let i = 0; i < len; i++){
    for(let k = i+1; k < len; k++){
      for(let j = k + 1; j < len; j++){
        const sum = nums[i] + nums[k] + nums[j];
        if (isPrime(sum)){
          answer += 1;
        }
      }
    }
  }
  
  return answer;
}

console.log(solution([1,2,3,4]));
console.log(solution([1,2,7,6,4]));

File: CS/PS/복습/progr_lv1/소수만들기.py
# 18:14 ~ 18:27
import itertools
import math
def is_prime(num):
    sqrt = int(math.sqrt(num))
    for i in range(2, sqrt + 1):
        if num % i == 0:
            return False
    return True
    
def solution(nums):
    answer = 0
    combis = list(map(list,itertools.combinations(nums, 3)))
    for combi in combis:
        if is_prime(sum(combi)):
            answer += 1
    return answer